pkg/helm/release/manager.go

```diff
@@ -145,12 +145,19 @@ func (m *manager) Sync(ctx context.Context) error {
 }
 
 func (m manager) syncReleaseStatus(status types.HelmAppStatus) error {
-	if status.Release == nil {
+	var release *rpb.Release
+	for _, condition := range status.Conditions {
+		if condition.Type == types.ConditionDeployed && condition.Status == types.StatusTrue {
+			release = condition.Release
```

Comment:
We might have had this conversation already so sorry if I'm rehashing but is there no way we can do without relying on the installed/latest release stored in the CR status?

I might be wrong but this seems like we're using the conditions as a state machine for our operator logic(to sync the storage backend to the latest release).

Is it possible to get the latest release for this CR directly from the server?

Reply:
Of course it's no different than what we were already doing with `status.Release` so not an issue with this PR but just something I'm wondering if we can address.

Reply:
Yeah, we did have that conversation, but I don't recall if we discussed this particular aspect of it.

My guess is that the syncing was done this way because we're using the tiller's memory storage driver ([here](https://github.com/operator-framework/operator-sdk/blob/d4a7b311ba8fa7536b3ce2289427041c488ddb71/test/helm-operator/cmd/helm-operator/main.go#L74) and [here](https://github.com/operator-framework/operator-sdk/blob/master/commands/operator-sdk/cmd/up/local.go#L209)) and this is necessary to bootstrap the operator if it has been restarted while there are active CRs. 

Generally, syncing would be a no-op because the releases are all already stored in the storage backend.

To solve the problem, we could switch to a ConfigMap or Secret implementation of the storage backend.

Reply:
Due to this, I think that the Release should potentially still be on the top level of the status IMO.

Reply:
We have a few options on where to put the release object:
1. In the condition, like it is now. The benefit here is that we can put the most recent successful release in the `Deployed` condition and the most recent failed release in the `ReleaseFailed` condition.
    - This could be helpful in the case where an install is successful and a subsequent update fails. In that situation, the previous installed release is still active.
    - It's unclear from the [status convention doc](https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#typical-status-properties) whether extra fields like this are allowed in the Condition. That may be an argument in favor of top-level fields.
2. In the top level, as you suggest. Would we put just the successful release or have top level fields for both successful and failed releases?
3. In a separate object entirely (which we may do anyway, based on my previous comment) with an object reference in the CR status either in the condition or at the top level, as suggested in the convention doc:
> Status information that may be large (especially proportional in size to collections of other resources, such as lists of references to other objects -- see below) and/or rapidly changing, such as resource usage, should be put into separate objects, with possibly a reference from the original object. This helps to ensure that GETs and watch remain reasonably efficient for the majority of clients, which may not need that data.

Reply:
> It's unclear from the status convention doc whether extra fields like this are allowed in the Condition. That may be an argument in favor of top-level fields.

I think you can add some but that they should be small,  What about just adding the Version, and in the top level of the status has only the active one. so say you have a failed upgrade from Y to X. The active condition would be a failure with the X release version (doesn't helm store this or is that point 3 above?) and the status object is for Y because that is actually still the active release?

Reply:
For the issue of not needing to rely on the status to sync the in memory storage driver, we can revisit this in a followup PR by most likely using the ConfigMap driver.

For this PR I would lean towards keeping the conditions as they are.
Once we're using a Configmap driver to store the release objects in a Configmap, then we can remove them from the conditions, and probably just store the release version, either in the condition or as a top level field as shawn suggested.
@joelanford WDYT?